### ISSUE TITLE
Failure sending parameters with Transaction::queryRows

### ISSUE THREAD

**Issue by cretz** [alaisi/postgres-async-driver#34]
For some strange reason, in 0.9, when I do `Transaction::queryRows` with something like `INSERT INTO TEST(FOO) VALUES($1) RETURNING ID` and I pass in the param, it fails with "ERROR: SQLSTATE=42P02, MESSAGE=there is no parameter $1". Yet, if I do `querySet` or outside of a transaction it does not. The reason for this is not obvious from looking at the code.

I am working on narrowing down a test case, but it is taking a while. Any ideas in the meantime?

**Comment by cretz** [comment]
Diff for test to replicate:

```diff
diff --git a/src/test/java/com/github/pgasync/impl/TransactionTest.java b/src/test/java/com/github/pgasync/impl/TransactionTest.java
index 3d8c3eb..902ec21 100644
--- a/src/test/java/com/github/pgasync/impl/TransactionTest.java
+++ b/src/test/java/com/github/pgasync/impl/TransactionTest.java
@@ -81,6 +81,17 @@ public class TransactionTest {
     }

     @Test
+    public void shouldCommitParameterizedInsertInTransaction() throws Exception {
+        long id = dbr.db().begin().flatMap(txn ->
+            txn.queryRows("INSERT INTO TX_TEST (ID) VALUES ($1) RETURNING ID", 35).first().flatMap(row -> {
+                Long value = row.getLong(0);
+                return txn.commit().map(v -> value);
+            })
+        ).toBlocking().single();
+        assertEquals(35L, id);
+    }
+
+    @Test
     public void shouldRollbackTransaction() throws Exception {
         CountDownLatch sync = new CountDownLatch(1);
```

**Comment by cretz** [comment]
PR to fix sent in #35
